fluent.runtime/fluent/runtime/resolver.py

```diff
@@ -256,44 +300,55 @@ def select_from_select_expression(self, env, key):
                 break
 
         if found is None:
+            if default is None:
+                env.errors.append(FluentFormatError("No default"))
+                return FluentNone()
             found = default
         return found.value(env)
 
 
-def is_number(val):
+def is_number(val: Any) -> bool:
     return isinstance(val, (int, float))
 
 
-def match(val1, val2, env):
+def match(val1: Any, val2: Any, env: ResolverEnvironment) -> bool:
     if val1 is None or isinstance(val1, FluentNone):
         return False
     if val2 is None or isinstance(val2, FluentNone):
         return False
     if is_number(val1):
         if not is_number(val2):
             # Could be plural rule match
-            return env.context._plural_form(val1) == val2
+            return cast(bool, env.context._plural_form(val1) == val2)
     elif is_number(val2):
         return match(val2, val1, env)
 
-    return val1 == val2
+    return cast(bool, val1 == val2)
```

Comment:
It's because the inputs are `Any`, and that at least technically allows for an `__eq__(self, other)` overload that returns something other than a `bool`.